server/monitor.go

```diff
@@ -3114,10 +3190,51 @@ func (s *Server) healthz(opts *HealthzOptions) *HealthStatus {
 	if opts == nil {
 		opts = &HealthzOptions{}
 	}
+	details := opts.Details
+	if details {
+		defer func() {
+			if len(health.Errors) != 0 {
+				health.Status = "error"
+			}
+		}()
+	}
+
+	if opts.Account == "" && opts.Stream != "" {
+		if !details {
+			health.Status = "error"
+			health.Error = fmt.Sprintf("Bad request: %q must not be empty when checking stream health", "account")
+		} else {
+			health.Errors = append(health.Errors, HealthzError{
+				Type:  HealthzErrorBadRequest,
+				Error: fmt.Sprintf("Bad request: %q must not be empty when checking stream health", "account"),
+			})
+		}
+		return health
```

Comment:
maybe add a Code field to the health result here? It would be nice to get an http 400 for these and a 404 for when the streams/consumers are not found for example

Reply:
or could base on the healthzError.Type to return a custom http code

Reply:
I'm not sure about the custom error code as this will work fine for http-based `HealthZ` only, and not for `$SYS.REQ.SERVER.%s.HEALTHZ`, unless we add a custom header. I will add a `StatusCode` field in response.